src/app/clusters/icd-management-server/ICDManagementCluster.cpp

```diff
@@ -109,141 +88,305 @@ void ICDManagementFabricDelegate::OnFabricRemoved(const FabricTable & fabricTabl
 {
-    return encoder.Encode(mICDConfigurationData->GetFeatureMap());
+    BitFlags<IcdManagement::Feature> featureMap = mICDConfigurationData.GetFeatureMap();
+    const bool hasCIP                           = featureMap.Has(Feature::kCheckInProtocolSupport);
+
+    if (hasCIP)
+    {
+        static constexpr CommandId kGeneratedCommands[] = {
+            Commands::RegisterClientResponse::Id,
+        };
+        ReturnErrorOnFailure(builder.ReferenceExisting(kGeneratedCommands));
+    }
+
+    if (mICDConfigurationData.GetFeatureMap().Has(Feature::kLongIdleTimeSupport) ||
+        mEnabledCommands.Has(OptionalCommands::kStayActive))
+    {
+        ReturnErrorOnFailure(builder.EnsureAppendCapacity(1));
+        ReturnErrorOnFailure(builder.Append(Commands::StayActiveResponse::Id));
+    }
+    return CHIP_NO_ERROR;
 }
 
 #if CHIP_CONFIG_ENABLE_ICD_LIT
-CHIP_ERROR ICDManagementAttributeAccess::ReadOperatingMode(EndpointId endpoint, AttributeValueEncoder & encoder)
+CHIP_ERROR ICDManagementCluster::ReadOperatingMode(AttributeValueEncoder & encoder)
 {
-    return mICDConfigurationData->GetICDMode() == ICDConfigurationData::ICDMode::SIT
+    return mICDConfigurationData.GetICDMode() == ICDConfigurationData::ICDMode::SIT
         ? encoder.Encode(IcdManagement::OperatingModeEnum::kSit)
         : encoder.Encode(IcdManagement::OperatingModeEnum::kLit);
 }
 #endif // CHIP_CONFIG_ENABLE_ICD_LIT
-
 #if CHIP_CONFIG_ENABLE_ICD_CIP
-CHIP_ERROR ICDManagementAttributeAccess::ReadRegisteredClients(EndpointId endpoint, AttributeValueEncoder & encoder)
+
+ICDManagementClusterWithCIP::ICDManagementClusterWithCIP(
+    EndpointId endpointId, PersistentStorageDelegate & storage, Crypto::SymmetricKeystore & symmetricKeystore,
+    FabricTable & fabricTable, ICDConfigurationData & icdConfigurationData, OptionalAttributeSet optionalAttributeSet,
+    BitMask<IcdManagement::OptionalCommands> aEnabledCommands,
+    BitMask<IcdManagement::UserActiveModeTriggerBitmap> aUserActiveModeTriggerBitmap, CharSpan aUserActiveModeTriggerInstruction) :
+    ICDManagementCluster(endpointId, storage, symmetricKeystore, fabricTable, icdConfigurationData, optionalAttributeSet,
+                         aEnabledCommands, aUserActiveModeTriggerBitmap, aUserActiveModeTriggerInstruction)
+{}
+
+CHIP_ERROR ICDManagementClusterWithCIP::Startup(ServerClusterContext & context)
 {
-    uint16_t supported_clients                    = mICDConfigurationData->GetClientsSupportedPerFabric();
-    PersistentStorageDelegate * storage           = mStorage;
-    Crypto::SymmetricKeystore * symmetricKeystore = mSymmetricKeystore;
-    const FabricTable * fabricTable               = mFabricTable;
+    // Initialize and register Fabric delegate
+    mFabricDelegate.Init(mStorage, &mSymmetricKeystore, mICDConfigurationData);
+    mFabricTable.AddFabricDelegate(&mFabricDelegate);
+    return CHIP_NO_ERROR;
```

Comment:
```suggestion
    return mFabricTable.AddFabricDelegate(&mFabricDelegate);
```

Since this returns a CHIP_ERROR. Although this is existing code, so maybe if we want bug-for-bug compatibility we can add just a comment on why we ignore the return value...